2018/2018_File_Generation_Features.md

```diff
@@ -0,0 +1,67 @@
+
+FilesGenerator also contains a method, `edit_report` that produces an edit report for a test file which is stored locally as a csv file. The edit report csv is stored in a directory and with a filename specified in `test_filepaths.yaml` 
+
+## The Large Test Files Script
+The Large Test Files script in [`large_test_files_script.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/large_test_files_script.py) creates a test file based on existing test file data to create a new file with a larger number of rows. The script would only need to be run, and the attributes of the new file to be created can be changed with [`large_file_script_config.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/large_file_script_config.yaml) in `python/configurations`. Files can be created with any number of rows, and can be made for any test institution so long as the Institution Name, LEI and Tax ID are provided in the configuration file. The configuration file stores filepaths for source data, as well as the filepaths for output. 
+
+In addition, the script contains functionality for changing values by column and row using a yaml configuration. An example of this yaml configuration is located in `python/configurations/` as 
+[`row_by_row_modification.yaml.`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/row_by_row_modification.yaml) The script can contain a path to a yaml configuration to initiate this functionality. 
+
+
+## Additional Utilities
+In addition to test file creation features, the repository contains utility functions that work with existing test file data. These functions assist in creating test files with further customization, such as files with a large number of rows or files as part of a new institution. The functions are located in [`python/utils.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/utils.py). 
+
+These functions include:
+- `write_file(path, ts_input, lar_input, name="test_file.txt")` 
+	Writes TS and LAR dataframes to a HMDA submission test file with a specified file path and directory. 
+
+- `read_data_frames(ts_df, lar_df)` 
+	Reads in a HMDA submission test file and outputs a TS and LAR dataframe. 
+
+- `unique_uli(new_lar_df, lei)`
+	Writes a set of unique ULI’s with check digits to a LAR dataframe given an LEI input. 
+
+- `new_lar_rows(row_count, lar_df, ts_df)` 
+	Duplicates or retracts rows in a LAR dataframe to produce a file with a certain number of rows. Provides capabilities for creating LAR dataframes with large row counts. 
+
+- `row_by_row_modification(lar_df, yaml_filepath='configurations/row_by_row_modification.yaml')`
+	Changes a LAR dataframe given changes specified in a yaml file. It allows users to test for minor changes in LAR data.
```

Comment:
This line is confusing to me.

Is the purpose of the method something like:
Creates test files with customizable record count and data variables for testing load and edge cases?

Reply:
Yes, I think that would be a better way to say it.